src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -50,50 +59,81 @@ class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
           return isValidSharePointUrl;
         }
 
-        const invalidIds = formatting
-          .splitAndTrim(args.options.ids)
-          .filter(id => !validation.isValidGuid(id));
-        if (invalidIds.length > 0) {
-          return `The following IDs are invalid: ${invalidIds.join(', ')}`;
+        if (args.options.ids) {
+          const invalidIds = formatting
+            .splitAndTrim(args.options.ids)
+            .filter(id => !validation.isValidGuid(id));
+
+          if (invalidIds.length > 0) {
+            return `The following IDs are invalid: ${invalidIds.join(', ')}`;
+          }
         }
 
         return true;
       }
     );
   }
 
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['ids', 'allPrimary', 'allSecondary'] });
```

Comment:
In the specs, we agreed that it should be possible to provide `allPrimary` and `allSecondary` together. With option sets, this isn't possible.